### ISSUE TITLE
Filters should be grouped using a <fieldset> and <legend> markup

### ISSUE THREAD

**Issue by mmiddaugh** [department-of-veterans-affairs/va.gov-cms#12881]
## Description
When first tabbing into the search filters, a screen reader user will hear "Use my location". However, with the addition of the visual cues, it is not immediately apparent what the purpose of this button is - what is the user using their location for? Providing additional context through a group label e.g. "Find a VA location" will make it clearer without the need of navigating around the button itself for more context.

[Codepen for demonstrating how a group label could work](https://codepen.io/joshkimux/pen/WNGmXxG)

## Acceptance Criteria
- [ ] _What will be created or happen as a result of this story?_
---

**Comment by laflannery** [comment]
After investigating further, closing this in favor of #12882. A button is not going to get the context from a legend as it's not a field element so we should provide context in a different way
